src/Tasks/Exec.cs

```diff
@@ -611,12 +611,26 @@ protected internal override void AddCommandLineCommands(CommandLineBuilderExtens
 
                     // If for some crazy reason the path has a & character and a space in it
                     // then get the short path of the temp path, which should not have spaces in it
-                    // and then escape the &
                     if (batchFileForCommandLine.Contains("&") && !batchFileForCommandLine.Contains("^&"))
                     {
                         batchFileForCommandLine = NativeMethodsShared.GetShortFilePath(batchFileForCommandLine);
-                        batchFileForCommandLine = batchFileForCommandLine.Replace("&", "^&");
                     }
+
+                    StringBuilder fileName = StringBuilderCache.Acquire(batchFileForCommandLine.Length);
+
+                    // Escape any '(', ')', or '&'
+                    for (int i = 0; i < batchFileForCommandLine.Length; i++)
+                    {
+                        char c = batchFileForCommandLine[i];
+
+                        if ((c == '(' || c == ')' || c == '&') && (i > 0 && batchFileForCommandLine[i - 1] != '^'))
```

Comment:
```suggestion
                        if ((c == '(' || c == ')' || c == '&') && (i == 0 || batchFileForCommandLine[i - 1] != '^'))
```

Reply:
For the record I can't think of a scenario that ever hits this when i == 0. I suppose users can set their temp path via env var, and could theoretically set it to `(...` where it should be local to whatever their working directory is I guess?

Taking it as a "likely never going to happen, but doesn't hurt" change.

Reply:
For the record—I don't disagree, so I wouldn't mind if you reduced it to `batchFileForCommandLine[i - 1] != '^'`.